File: katas/leetcode/771.Jewels_And_Stones.java
class Solution {
    public int numJewelsInStones(String J, String S) {
        Set<Character> jewels = new HashSet<>();
        int count = 0;

        for(char jewel : J.toCharArray()) {
            jewels.add(jewel);
        }

        for (char stone : S.toCharArray()) {
            if (jewels.contains(stone)) { count++; }
        }

        return count;
    }
}

File: katas/leetcode/771.Jewels_And_Stones.rs
impl Solution {
    pub fn num_jewels_in_stones(jewels: String, stones: String) -> i32 {
        let mut counter = 0;

        for stone in stones.chars() {
            if jewels.contains(stone) {
                counter += 1;
            }
        }

        return counter;
    }
}
